### ISSUE TITLE
[v6r15] Error in FileCatalogClient

### ISSUE THREAD

**Issue by andresailer** [DIRACGrid/DIRAC#2962]
```
self = <DIRAC.Resources.Catalog.FileCatalogClient.FileCatalogClient object at 0x7f80434f68d0>, path = {'myslcio.slcio': True}, timeout = 120

    @checkCatalogArguments
    def getFileUserMetadata( self, path, timeout = 120 ):
      """Get the meta data attached to a file, but also to
        the its corresponding directory
        """
>     directory = "/".join( path.split( "/" )[:-1] )
E     AttributeError: 'dict' object has no attribute 'split'
```
We get an error when calling getFileUserMetadata, the checkCatalogArguments decorator turns the path argument into a dictionary, right? But the function itself assumes path is a string.
So is that decorator there incorrectly, or does the function need a fix?
Removing the decorator does allow our unit test to pass again, but it is not clear what is meant to be correct.

**Comment by chaen** [comment]
Naively I would say that the function need a fix. checkCatalogArguments make the the input standard, so the function should accept standard arguments, that is a dict

**Comment by andresailer** [comment]
Hmm, ```getDirectoryUserMetadata``` doesn't have the decorator...
I hope @atsareg will know

**Comment by atsareg** [comment]
Yes, clearly overlooked, thanks for spotting

**Comment by andresailer** [comment]
So what is the problem here: is the decorator too much, does the function need to be adapted?
Will this be fixed soon so we can go on with our validation? If someone tells us what the correct behaviour should be we can try fixing this ourselves.

**Comment by atsareg** [comment]
I am fixing it to the next v6r15 patch

**Comment by andresailer** [comment]
Ok, thanks.

**Comment by andresailer** [comment]
This is not fixed in v6r15p3, is it?

**Comment by atsareg** [comment]
not yet

**Comment by atsareg** [comment]
Should be fixed in #2991. I wanted to make a proper service bulk call, but this is not exactly trivial, especially for a patch. So, a simple fix for the moment.

**Comment by fstagni** [comment]
Released (should be v6r15p4).
